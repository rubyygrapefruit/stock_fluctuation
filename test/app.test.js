const request = require("supertest");
const app = require("../server/app.js").app;
const mongoose = require("../server/app.js").mongoose;

afterAll(() => {
  mongoose.connection.close();
});

describe("Test the root path", () => {
  test("It should response the GET method", done => {
    request(app)
      .get("/")
      .then(response => {
        expect(response.statusCode).toBe(200);
        done();
      });
  });
});

describe("Test restful api", () => {
  test("It should respond with data length equal to 4", done => {
    request(app)
      .get("/api")
      .then(res => res.body)
      .then(data => {
        expect(data.length).toBe(4);
        done();
      });
  });

  test("It should respond with an array of objects", done => {
    request(app)
      .get("/api")
      .then(res => res.body)
      .then(data => {
        expect(Array.isArray(data)).toBe(true);
        done();
      });
  });
});
