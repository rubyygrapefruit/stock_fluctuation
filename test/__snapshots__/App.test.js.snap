// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render a <div /> 1`] = `
<div
  className="uk-container-small"
>
  <Header />
  <GraphHeaderContainer
    currentPrice={null}
    lastPrice={null}
  />
  <Graph
    allTicks={Array []}
    currentTicks={Array []}
    onUpdatePrice={[Function]}
  />
  <Footer
    changeTime={[Function]}
  />
</div>
`;
